src/coreclr/tests/src/CLRTest.CrossGen.targets

```diff
@@ -72,55 +72,68 @@ if [ ! -z ${RunCrossGen2+x} ]%3B then
       fi
     fi
 
-    __OutputFile="$PWD/"
-    if [ ! -z ${CompositeBuildMode+x} ]%3B then
-        __OutputFile+="composite-r2r.dll"
-    else
-        __OutputFile+="$(MSBuildProjectName).dll"
+    if [ ! -z ${LargeVersionBubble+x} ]%3B then
+        ExtraCrossGen2Args+=" --inputbubble"
     fi
 
-    __ResponseFile="$__OutputFile.rsp"
-    rm $__ResponseFile
+    __cg2ExitCode=0
 
-    __Command=$_DebuggerFullPath
-    # Tests run locally need __TestDotNetCmd (set by runtest.py) or a compatible 5.0 dotnet runtime in the path
-    if [ ! -z ${__TestDotNetCmd+x} ] %3B then
-        __Command+=" $__TestDotNetCmd"
-    else
-        __Command+=" dotnet"
-    fi
-    __Command+=" $CORE_ROOT/crossgen2/crossgen2.dll"
-    __Command+=" @$__ResponseFile"
-    __Command+=" $ExtraCrossGen2Args"
+   OneFileCrossgen2() {
+      __OutputFile=$1
+
+      __ResponseFile="$__OutputFile.rsp"
+      rm $__ResponseFile
+
+      __Command=$_DebuggerFullPath
+      # Tests run locally need __TestDotNetCmd (set by runtest.py) or a compatible 5.0 dotnet runtime in the path
+      if [ ! -z ${__TestDotNetCmd+x} ] %3B then
+          __Command+=" $__TestDotNetCmd"
+      else
+          __Command+=" dotnet"
+      fi
+      __Command+=" $CORE_ROOT/crossgen2/crossgen2.dll"
+      __Command+=" @$__ResponseFile"
+      __Command+=" $ExtraCrossGen2Args"
+
+      echo $2 >> $__ResponseFile
+
+      echo -o:$__OutputFile>>$__ResponseFile
+      echo -r:$CORE_ROOT/System.*.dll>>$__ResponseFile
+      echo -r:$CORE_ROOT/Microsoft.*.dll>>$__ResponseFile
+      echo -r:$CORE_ROOT/mscorlib.dll>>$__ResponseFile
+      echo --targetarch:$(TargetArchitecture)>>$__ResponseFile
+      echo -O>>$__ResponseFile
+
+      echo "Response file: $__ResponseFile"
+      cat $__ResponseFile
+      echo "Running CrossGen2: $__Command"
+      $__Command
+      __cg2ExitCode=$?
+  }
 
     if [ ! -z ${CompositeBuildMode+x} ]%3B then
-      echo --composite>>$__ResponseFile
-      echo $PWD/IL/*.dll>>$__ResponseFile
+        ExtraCrossGen2Args+= --composite
+        OneFileCrossgen2 "$PWD/composite-r2r.dll" "$PWD/IL/*.dll"
     else
-      echo --inputbubble>>$__ResponseFile
-      echo $PWD/IL/$(MSBuildProjectName).dll>>$__ResponseFile
-      echo -r:$PWD/IL/*.dll>>$__ResponseFile
+        ExtraCrossGen2Args+= -r:$PWD/IL/*.dll
+      for dllFile in $PWD/IL/*.dll
+      do
+        echo $dllFile
+        bareFileName="${dllFile##*/}"
+        OneFileCrossgen2 "$PWD/$bareFileName" "$dllFile"
+        if [ $__cg2ExitCode -ne 0 ]; then
```

Comment:
Should we be tolerant towards the exit code 2 you introduced for non-loadable inputs? At least my understanding is that the new exit code is supposed to cater for the cases where we blindly submit a native DLL to CG2.

Reply:
Eh, on Unix there aren't any files with the .dll extension that should have this characteristic.

Reply:
I see, I didn't realize that native libraries would get automatically filtered out on the other OS'es due to different extensions. Thanks for clarifying!